src/query/service/src/interpreters/access/privilege_access.rs

```diff
@@ -937,6 +948,9 @@ impl AccessChecker for PrivilegeAccess {
 
             }
             Plan::RenameTable(plan) => {
+                if  self.ctx.is_temp_table(&plan.catalog,&plan.database, &plan.table) {
```

Comment:
If a user does not  have any privilege, could it drop/truncate/alter the temp table?

Maybe you could create a new user that without any privilege and then test on local with these test.
![image](https://github.com/user-attachments/assets/a187cced-5766-411f-92b6-cb984466fa40)

Reply:
When drop/truncate/alter the temp table, it should pass the table privilege check here: https://github.com/SkyFan2002/databend/blob/dc8519184446c877a15ca3ebe0cb8adb7c930a95/src/query/service/src/interpreters/access/privilege_access.rs#L296

Reply:
> When drop/truncate/alter the temp table, it should pass the table privilege check here: https://github.com/SkyFan2002/databend/blob/dc8519184446c877a15ca3ebe0cb8adb7c930a95/src/query/service/src/interpreters/access/privilege_access.rs#L296

Ok. Now about privilege LGTM. Please add some privilege check about tmp table.